### PR TITLE
User listing

### PR DESCRIPTION
**Pull request by srkirve77**

### CODE REVIEW

File: src/tablebody/tablebody.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import UserList from'./tablebody'
+import { shallow, render, mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import enzyme from 'enzyme';
+import { TableBody } from "@material-ui/core";
+
+enzyme.configure({ adapter: new Adapter() })
```
Comment (Santosh-Punase) [inline src/tablebody/tablebody.test.js]: we have already configured adapter in test-setup

File: src/tablebody/tablebody.test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     const users = [ {"id":1, "name":"john"}]
+
+     let tree;
+     beforeEach(() => {
+          jest.clearAllMocks();
+          tree = shallow(<UserList pageNo = {pageNo} getTotalPages = {getTotalPages}/>);
+     }),
+
+     it('getUsers should have been called',()=>{
+          expect(tree.find(TableBody).text()).toHaveLength(0);
```
Comment (Santosh-Punase) [inline src/tablebody/tablebody.test.js]: Pls avoid false tests

File: src/server-conn/fetchUsers.jsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const fetchUsers = async(pageNo) => {
+    const res = await fetch('https://gorest.co.in/public-api/users?page='+pageNo,{})
+    const data = await res.json()
```
Comment (Santosh-Punase) [inline src/server-conn/fetchUsers.jsx:3]: is `await` needed for `line 3`

File: src/pagination/pagination.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  });
+
+const Pagination = ({onPage, currentPage, totalPages}) => {
+
+    const [page, setPage] = useState(1)
+    const classes = useStyles()
+
+    const getPage = () => {
+        if(isNaN(page))
```
Comment (Santosh-Punase) [inline src/pagination/pagination.jsx]: can this ever be `NaN`?

File: package.json [hunk 5]
```diff
@@ -2,20 +2,42 @@
...
+    "@babel/runtime": "^7.12.1",
+    "@material-ui/core": "^4.11.4",
+    "@material-ui/icons": "^4.11.2",
     "@testing-library/jest-dom": "^5.12.0",
     "@testing-library/react": "^11.2.6",
     "@testing-library/user-event": "^12.8.3",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
+    "chai": "^4.3.4",
```
Comment (Santosh-Punase) [inline package.json]: are we using `chai`?

File: src/tablebody/tablebody.test.js (hunk 1 above)
Comment (srkirve77) [reply src/tablebody/tablebody.test.js]: yeah sir, I will remove this adpator

**Review by Santosh-Punase — CHANGES_REQUESTED** [review]
Pls do the required changes

File: src/tablebody/tablebody.test.js (hunk 2 above)
Comment (srkirve77) [reply src/tablebody/tablebody.test.js]: Good point sir.noted.

File: src/pagination/pagination.jsx (hunk 4 above)
Comment (srkirve77) [reply src/pagination/pagination.jsx]: No sir.I will remove this block

File: package.json (hunk 5 above)
Comment (srkirve77) [reply package.json]: Yes sir.I was using chai but then in code i just switched to default expect.i will remove this

File: src/server-conn/fetchUsers.jsx (hunk 3 above)
Comment (srkirve77) [reply src/server-conn/fetchUsers.jsx:3]: yes sir await is needed here.without it `data` variable is not getting any value assigned to it

**Review by Santosh-Punase — DISMISSED** [review]

**Pull request merged**